backend/s3_backend.go

```diff
@@ -161,6 +165,15 @@ func (s *S3Backend) Open(db, version, file string) (io.ReadCloser, error) {
 	}
 	resp, err := s.svc.GetObject(params)
 
+	// If the download failed, retry maxRetries number of times with an
```

Comment:
I feel like we should check to see what the error is, before we start retrying. If we are getting a 400 level response code from s3 we should not retry.  Also adding a metric here for s3 retries might be a good idea.